app.py

```diff
@@ -1579,6 +1710,46 @@ def gpt_generate_room_title(messages):
 
     title = response.choices[0].message.content
     return title.replace('"', "")
+    
+    
+############################################################ C WUZ HERE
+def ollama_generate_room_title(messages):
+    """
+    Generate a title for the room based on a list of messages.
+    """
+    #openai_client, model_name = get_openai_client_and_model()
+    model_name = 'thewindmom/hermes-3-llama-3.1-8b'
+    chat_history = [
+        {
+            "role": "system" if msg.username in system_users else "user",
+            "content": f"{msg.username}: {msg.content}",
+        }
+        for msg in reversed(messages)
+        if not msg.is_base64_image()
+    ]
+
+    chat_history.append(
+        {
+            "role": "system",
+            "content": "return a short title for the title bar of this conversation. Reply with the title only",
+        }
+    )
+
+    # Interaction with LLM to generate summary
+    # For example, using OpenAI's GPT model
+    response = ollama.chat(
+        messages=chat_history,
+        model=model_name,  # or any appropriate model
+        #max_tokens=20,
+    )
+    
+    print(response)
+    content = response["message"]["content"]
+    title = content.strip().replace('"', "")
+
+    return title
+
+#############################################################################
```

Comment:
If we integrate ollama with `get_openai_client_and_model` like the other models we won't need a separate function for this work. Less is more.